src/test/java/com/webank/weid/full/evidence/TestCreateEvidence.java

```diff
@@ -649,4 +651,32 @@ public void testEmptyEvidenceAll() {
         // Any second attempt will fail
         Assert.assertFalse(evidenceService.setHashValue(hashValue, eviAddr, privKey).getResult());
     }
+
+    /**
+     * Case: generate hash value.
+     */
+    @Test
+    public void testGenerateHash() throws Exception {
+        // Credential and CredentialPojo
+        Assert.assertTrue(evidenceService.generateHash(credential).getResult().getHash()
+            .equalsIgnoreCase(credential.getHash()));
+        Assert.assertTrue(evidenceService.generateHash(credentialPojo).getResult().getHash()
+            .equalsIgnoreCase(credentialPojo.getHash()));
+        Assert.assertTrue(evidenceService.generateHash(selectiveCredentialPojo).getResult()
+            .getHash().equalsIgnoreCase(selectiveCredentialPojo.getHash()));
+        // Test file
```

Comment:
Please add comments to describe the file encoding that you are testing.

Reply:
done